agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -1484,6 +1484,52 @@ bool monitor_thread::hal_event_handler(bwl::base_wlan_hal::hal_event_ptr_t event
 
         // If event == Channel_Scan_New_Results_Ready do nothing since is_dump's default is 0
         if (event == Event::Channel_Scan_Dump_Result) {
+            auto msg = static_cast<bwl::sCHANNEL_SCAN_RESULTS_NOTIFICATION *>(data);
+
+            auto in_result  = &(msg->channel_scan_results);
+            auto out_result = &(notification->scan_results());
+
+            // Strings, simple arrays & vectors:
+            string_utils::copy_string(out_result->ssid, in_result->ssid,
+                                      beerocks::message::WIFI_SSID_MAX_LENGTH);
+            std::copy_n(in_result->bssid.oct, sizeof(out_result->bssid.oct), out_result->bssid.oct);
+            std::copy(in_result->basic_data_transfer_rates_kbps.begin(),
+                      in_result->basic_data_transfer_rates_kbps.end(),
+                      out_result->basic_data_transfer_rates_kbps);
+            std::copy(in_result->supported_data_transfer_rates_kbps.begin(),
+                      in_result->supported_data_transfer_rates_kbps.end(),
+                      out_result->supported_data_transfer_rates_kbps);
+
+            // Simple values
```

Comment:
The comment is unclear - what are simple values?

Reply:
Non-STL types?

Reply:
I'm changing the title to "Arrays"

`ssid` is a string that needs to be set in a `char[]`
`bssid` is a char array (mac) that needs to be copied
`basic_data_transfer_rates_kbps` & `supported_data_transfer_rates_kbps` are vectors that need to be set in an array
since all of them are basically arrays I think it's a better title

Reply:
Signed & unsigned integers
Changing it to "Primary types"